### PR TITLE
Test bells-1.0 with checkpoint/restore

### PR DESCRIPTION
**Pull request by anjumfatima90**
#23276

### CODE REVIEW

**Comment by anjumfatima90** [comment]
#build

File: dev/io.openliberty.checkpoint_fat/publish/servers/checkpointFaces-beanServer/bootstrap.properties [hunk 1]
```diff
@@ -13,4 +13,4 @@
 bootstrap.include=../testports.properties
 io.openliberty.checkpoint.dump.threads=true
 com.ibm.ws.logging.trace.specification=*=info:checkpoint=all
-io.openliberty.checkpoint.allowed.features=timedexit-1.0,concurrent-1.0,componenttest-1.0,concurrent-2.0,componenttest-2.0
+io.openliberty.checkpoint.allowed.features=concurrent-1.0,concurrent-2.0
```
Comment (tjwatson) [inline dev/io.openliberty.checkpoint_fat/publish/servers/checkpointFaces-beanServer/bootstrap.properties]: @dazavala is concurrent needed here because it became part of EE 10 web profile?  Not sure why these features would not have the header to enable checkpoint.

File: dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/BellsTest.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        String name = getTestMethodNameOnly(testName);
+        switch (name) {
+            case "testBellsCheckpointAtDeployment":
+                server.setCheckpoint(CheckpointPhase.DEPLOYMENT, false, null);
+                break;
+            case "testBellsCheckpointAtApplication":
+                server.setCheckpoint(CheckpointPhase.APPLICATIONS, false, null);
+                break;
+            default:
+                server.setCheckpoint(CheckpointPhase.APPLICATIONS, true, null);
```
Comment (tjwatson) [inline dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/BellsTest.java:79]: I prefer to fail here instead.

File: dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/WebProfileEARtest.java [hunk 3]
```diff
@@ -74,7 +74,6 @@ public void testEARdeployment() throws Exception {
 
         server.setCheckpoint(CheckpointPhase.APPLICATIONS, false, null);
         server.startServer();
-        server.stopServer();
```
Comment (tjwatson) [inline dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/WebProfileEARtest.java:77]: How is this related to the bells checkpoint support?  Looks like it is correct that the `stopServer` should not be needed here in this test though, so not requesting you change the PR for this.

File: dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/BellsTest.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
+/*******************************************************************************
+ * Copyright (c) 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
```
Comment (tjwatson) [inline dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/BellsTest.java]: Need to update this to use EPL-2.0:

```java
/*******************************************************************************
 * Copyright (c) 2023 IBM Corporation and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-2.0/
 * 
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     IBM Corporation - initial API and implementation
 *******************************************************************************/
```

File: dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/BellsTest.java [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    public static final String APP_NAME = "app1";
+
+    public static final String SERVER_NAME = "checkpointBells";
+
+    @Server(SERVER_NAME)
+    @TestServlet(servlet = TestServletA.class, contextRoot = APP_NAME)
+    public static LibertyServer server;
+
+    @ClassRule
+    public static RepeatTests repeatTest = MicroProfileActions.repeat(SERVER_NAME, TestMode.FULL, //
```
Comment (dazavala) [inline dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/BellsTest.java]: Did you intend to make a common on line 52?

File: dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/BellsTest.java (hunk 2 above)
Comment (anjumfatima90) [reply dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/BellsTest.java:79]: I added an enum to handle unknown test.

File: dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/WebProfileEARtest.java (hunk 3 above)
Comment (anjumfatima90) [reply dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/WebProfileEARtest.java:77]: Yes this is not related to bells. I came accross this while updating the bootstrap properties.

File: dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/BellsTest.java (hunk 4 above)
Comment (anjumfatima90) [reply dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/BellsTest.java]: Updated licence. Thanks!

File: dev/io.openliberty.checkpoint_fat/publish/servers/checkpointFaces-beanServer/bootstrap.properties (hunk 1 above)
Comment (anjumfatima90) [reply dev/io.openliberty.checkpoint_fat/publish/servers/checkpointFaces-beanServer/bootstrap.properties]: concurrent-2.0 has instant on enabled. May be I should just enable cocurrent-1.0 as well.

Comment (dazavala) [reply dev/io.openliberty.checkpoint_fat/publish/servers/checkpointFaces-beanServer/bootstrap.properties]: The Faces test committed two weeks before the InstantOn feature design change.  So, the property is obsolete now.   IDR why I added concurrent-x.0 to the test server; it may be unnecessary.  I'll determine whether concurrent-x.0 can be removed.

Comment (anjumfatima90) [reply dev/io.openliberty.checkpoint_fat/publish/servers/checkpointFaces-beanServer/bootstrap.properties]: Enabled instant on for concurrent-1.0 and updated the bootstrap properties in FAT.

**Review by dazavala — DISMISSED** [review]
@anjumfatima90 The BELL setup and usage LGTM.  Approved, though I have a conceptual question about test which Tom has already reviewed.

File: dev/com.ibm.websphere.appserver.features/visibility/public/concurrent-1.0/com.ibm.websphere.appserver.concurrent-1.0.feature [hunk 6]
```diff
@@ -23,3 +23,5 @@ Subsystem-Name: Concurrency Utilities for Java EE 1.0
 kind=ga
 edition=core
 WLP-Activation-Type: parallel
+WLP-InstantOn-Enabled: true
```
Comment (tjwatson) [inline dev/com.ibm.websphere.appserver.features/visibility/public/concurrent-1.0/com.ibm.websphere.appserver.concurrent-1.0.feature:26]: I think I am ok adding this here as long as we test concurrent-1.0 with checkpoint.  I think we do test it for EE 10 because it is included in EE 10 webprofile.

**Comment by anjumfatima90** [comment]
#build

**Comment by anjumfatima90** [comment]
#build

File: dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/FATSuite.java [hunk 7]
```diff
@@ -66,7 +66,8 @@
                 SlowAppStartTest.class,
                 JsonbTest.class,
                 JsonpTest.class,
-                ManagedBeansTest.class
+                ManagedBeansTest.class,
```
Comment (tevans78) [inline dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/FATSuite.java:69]: Please remember to update the copyright date in the header of any file which has functionally changed (any change except whitespace or formatting).

Comment (anjumfatima90) [reply dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/FATSuite.java:69]: Done, thanks!

File: dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/BellsTest.java [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+    public void setUp() throws Exception {
+        testMethod = getTestMethod(TestMethod.class, testName);
+        switch (testMethod) {
+            case testBellsCheckpointAtDeployment:
+                server.setCheckpoint(CheckpointPhase.DEPLOYMENT, false, null);
+                break;
+            case testBellsCheckpointAtApplication:
+                server.setCheckpoint(CheckpointPhase.APPLICATIONS, false, null);
+                break;
+            case testHttpServletRequest:
```
Comment (tjwatson) [inline dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/BellsTest.java:78]: This test case does not appear to be written yet.  Curious why you pass true here for auto-restore, but since there is no test method named testHttpServletRequest it is impossible to know.

Comment (anjumfatima90) [reply dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/BellsTest.java:78]: testHttpServletRequest is in app1. I am using an existing test app `app1`. The test is in `TestServletA` class.

**Review by tjwatson — APPROVED** [review]

**Comment by anjumfatima90** [comment]
#libby
